src/commands/mod.rs

```diff
@@ -3142,6 +3150,24 @@ fn cmd_workspace_list(
     Ok(())
 }
 
+fn cmd_workspace_root(
+    ui: &mut Ui,
+    command: &CommandHelper,
+    _args: &WorkspaceRootArgs,
+) -> Result<(), CommandError> {
+    let workspace_command = command.workspace_helper(ui)?;
+    let root =
+        workspace_command
+            .workspace_root()
+            .to_str()
+            .ok_or_else(|| CommandError::UserError {
+                message: String::from("The workspace root is not valid UTF-8"),
+                hint: None,
+            })?;
+    writeln!(ui, "{root}",)?;
```

Comment:
nit: excessive comma for oneliner

Reply:
Indeed. `cargo fmt` let it untouched because it belongs to a macro invocation.